### PR TITLE
Update contributer guide

### PR DESCRIPTION
**Pull request by nikclayton**
- How to find issues to work on
- Best practices for issues and PRs
- Describe linting
- Describe project names for GitHub issue management

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,3 +84,67 @@ Tusky is primarily a Mastodon client and aims to always support the newest Masto
...
+
+### How issues are assigned
+
+1. An issue is never assigned *by* someone *to* someone else. As a volunteer project the assigner can never be sure that the assignee is available to resolve the issue right now.
+
+2. Therefore, **issues are only ever self-assigned**.
+
+3. Self-assigning an issue is a clear statement of “I will do the work necessary to resolve this issue. If I can not do that in reasonable time I will un-assign the issue”
+
+4. If you have assigned an issue to yourself you are expected to provide periodic updates to that issue, at least one update every 7 days. This is to ensure that the state of the work on the issue is clear to everyone else on the project.
```
[the remark this answers is not in the document]
Comment (nikclayton) [reply CONTRIBUTING.md:100]: Yes, it's from the meeting.

If you can't provide an update every 7 days for an issue you've assigned to yourself then just assign yourself from the issue. Maybe leave a comment if you're going away for two weeks, and explain whether you want someone else to pick it up, or you'll get back to it when you return.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,3 +84,67 @@ Tusky is primarily a Mastodon client and aims to always support the newest Masto
...
+
+    a. The update may be a variation of “I have not been able to work on this”. That's OK. The important thing is that this is valuable information to convey to everyone else.
+
+    b. If a PR is associated with the issue then updates to the PR are treated as updates to the issue
+
+5. It is OK if you have assigned an issue to yourself, and then discover that you are not able to complete it or provide useful updates. If that happens just un-assign the issue. You don't have to provide a justification for why you can't do it at this time. It's enough to let the rest of the team know that someone else will need to pick this up.
+
+6. If you are working on an issue and someone else needs to make a decision, `@` them into the issue, and add the `blocked` label so they can find it.
```
[the remark this answers is not in the document]
Comment (nikclayton) [reply CONTRIBUTING.md:108]: Ask in the chat channel first? Or @- mention several people in to the issue?

If you've assigned yourself to an issue, you're blocked, you've escalated, and no one is responding it's also OK to unassign yourself from the issue.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -17,14 +17,47 @@ To add a new language, click on the 'Start a new translation' button on at the b
...
+### Find something to work on
+
+https://github.com/tuskyapp/Tusky/contribute is a list of open issues that have been tagged `good first issue`. This means the changes should be relatively small and self-contained.
+
+You do not have to restrict yourself to just working on one of those issues, but as the name suggests, they're a good place to start.
+
+### Best practices for working on an issue
+
+- Post a comment to the issue, saying that you plan on working on it
+- Join the developer chat room, https://riot.im/app/#/room/#Tusky:matrix.org,
```
Comment (nikclayton) [inline CONTRIBUTING.md:29]:
```suggestion
- Join the developer chat room, https://matrix.to/#/#Tusky-General:matrix.org, 
```

[the remark this answers is not in the document]
Comment (nikclayton) [reply CONTRIBUTING.md]: Fixed, thanks.

**Comment by mcclure** [comment]
From discussion in 2023-Aug-15 contributor meeting: Brought up argument (see Lakoja comment above) 7 days is too short given the commitment level of most contributors, majority at meeting agreed with this. Nik agreed to investigate whether a GH action can (1) after 7 days without updates, remind the to either post an update or remove themselves as issue owner and (2) after 14 days, remove the issue owner.

If this is possible, I believe the meeting was okay with 14 days being the "must post or deassign" limit, and with a soft expectation that people post at least every 7 days an issue is assigned to them if they are able (even if the update is to say "no time to update") .

**Pull request closed without merging**